lib/rules/header.js

```diff
@@ -38,23 +73,32 @@ module.exports = function(context) {
     // If commentType is line then we want an array of the lines,
     // but if block then we want just a string
     var header, headerLines;
+    var shouldFix = false;
```

Comment:
Would this be better named `canFix`?

Either way, could you add a brief comment explaining that this is false if any lines are regexes?